File: quiz_app_flutter/lib/domain/repository/quiz_repository.dart
import '../../core/resource.dart';
import '../entities/dashboard/dashboard_data.dart';
import '../entities/quiz/quiz_data.dart';
import '../entities/quiz_set/quiz_set_data.dart';

abstract class QuizRepository {
  Stream<Resource<DashboardData>> getDashboardData();

  Stream<Resource<QuizSetData>> getQuizListByTopic(String topic);

  Stream<Resource<List<QuizData>>> getQuizzesBySetAndTopic(String fileName);
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/domain/repositry/QuizRepository.kt
package com.tnm.quizappkmp.domain.repositry

import com.tnm.quizappkmp.domain.model.Resource
import com.tnm.quizappkmp.domain.model.dashboard.DashboardData
import com.tnm.quizappkmp.domain.model.quiz.QuizData
import com.tnm.quizappkmp.domain.model.quizSets.QuizSetData
import kotlinx.coroutines.flow.Flow

interface QuizRepository {
    fun getDashboardData(): Flow<Resource<DashboardData>>
    fun getQuizListByTopic(topic: String): Flow<QuizSetData?>
    fun getQuizzesBySetAndTopic(fileName: String): Flow<Resource<List<QuizData>>>

}